In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/daily-covid-19-data-2020-2024/WHO-COVID-19-global-data.csv
/kaggle/input/daily-covid-19-data-2020-2024/Global COVID-19 Data (2020-2024)/Global COVID-19 Data (2020-2024)/WHO-COVID-19-global-table-data.csv
/kaggle/input/daily-covid-19-data-2020-2024/Global COVID-19 Data (2020-2024)/Global COVID-19 Data (2020-2024)/vaccination-metadata.csv
/kaggle/input/daily-covid-19-data-2020-2024/Global COVID-19 Data (2020-2024)/Global COVID-19 Data (2020-2024)/WHO-COVID-19-global-data.csv
/kaggle/input/daily-covid-19-data-2020-2024/Global COVID-19 Data (2020-2024)/Global COVID-19 Data (2020-2024)/vaccination-data.csv


In [2]:
# Load the CSV files
daily_covid_data = pd.read_csv('/kaggle/input/daily-covid-19-data-2020-2024/Global COVID-19 Data (2020-2024)/Global COVID-19 Data (2020-2024)/WHO-COVID-19-global-data.csv', sep=';')  # Assuming semicolon delimiter
vaccination_data = pd.read_csv('/kaggle/input/daily-covid-19-data-2020-2024/Global COVID-19 Data (2020-2024)/Global COVID-19 Data (2020-2024)/vaccination-data.csv', sep=';')

---------------

# Data Cleaning of: WHO-COVID-19-global-data

In [3]:
daily_covid_data.head(20)

Date_reported Country_code      Country WHO_region  New_cases  \
0     05/01/2020           AF  Afghanistan       EMRO        NaN   
1     12/01/2020           AF  Afghanistan       EMRO        NaN   
2     19/01/2020           AF  Afghanistan       EMRO        NaN   
3     26/01/2020           AF  Afghanistan       EMRO        NaN   
4     02/02/2020           AF  Afghanistan       EMRO        NaN   
5     09/02/2020           AF  Afghanistan       EMRO        NaN   
6     16/02/2020           AF  Afghanistan       EMRO        NaN   
7     23/02/2020           AF  Afghanistan       EMRO        NaN   
8     01/03/2020           AF  Afghanistan       EMRO        1.0   
9     08/03/2020           AF  Afghanistan       EMRO        NaN   
10    15/03/2020           AF  Afghanistan       EMRO        6.0   
11    22/03/2020           AF  Afghanistan       EMRO       17.0   
12    29/03/2020           AF  Afghanistan       EMRO       67.0   
13    05/04/2020           AF  Afghanistan       EMRO      183.0   
14    12/04/2020           AF  Afghanistan       EMRO      247.0   
15    19/04/2020           AF  Afghanistan       EMRO      387.0   
16    26/04/2020           AF  Afghanistan       EMRO      422.0   
17    03/05/2020           AF  Afghanistan       EMRO      841.0   
18    10/05/2020           AF  Afghanistan       EMRO     1392.0   
19    17/05/2020           AF  Afghanistan       EMRO     2490.0   

    Cumulative_cases  New_deaths  Cumulative_deaths  
0                  0         NaN                  0  
1                  0         NaN                  0  
2                  0         NaN                  0  
3                  0         NaN                  0  
4                  0         NaN                  0  
5                  0         NaN                  0  
6                  0         NaN                  0  
7                  0         NaN                  0  
8                  1         NaN                  0  
9                  1         NaN                  0  
10                 7         NaN                  0  
11                24         NaN                  0  
12                91         2.0                  2  
13               274         3.0                  5  
14               521        10.0                 15  
15               908        15.0                 30  
16              1330        13.0                 43  
17              2171        21.0                 64  
18              3563        41.0                105  
19              6053        48.0                153

In [4]:
daily_covid_data.isna().sum()

Date_reported            0
Country_code           241
Country                  0
WHO_region            4338
New_cases            18812
Cumulative_cases         0
New_deaths           32839
Cumulative_deaths        0
dtype: int64

In [5]:
daily_covid_data.shape

(57840, 8)

In [6]:
daily_covid_data = daily_covid_data.dropna(subset=['Country_code'])

- Dropped Null values in column 'Country_code' cause it has 241 Null values out of 57840 rows

In [7]:
daily_covid_data['WHO_region'] =daily_covid_data['WHO_region'].fillna('unknown')

- filled every unknown region with 'unknown'

In [8]:
daily_covid_data.isna().sum()

Date_reported            0
Country_code             0
Country                  0
WHO_region               0
New_cases            18769
Cumulative_cases         0
New_deaths           32723
Cumulative_deaths        0
dtype: int64

In [9]:
daily_covid_data.describe().T

count          mean           std      min     25%  \
New_cases          38830.0  1.997797e+04  2.714684e+05 -65079.0    43.0   
Cumulative_cases   57599.0  1.799611e+06  7.813230e+06      0.0  4115.5   
New_deaths         24876.0  2.835775e+02  1.217300e+03  -3432.0     4.0   
Cumulative_deaths  57599.0  2.008246e+04  8.202777e+04      0.0    28.0   

                       50%       75%          max  
New_cases            394.5    4015.0   40475477.0  
Cumulative_cases   45852.0  531723.5  103436829.0  
New_deaths            20.0     106.0      47687.0  
Cumulative_deaths    552.0    7118.0    1194158.0

In [10]:
daily_covid_data.duplicated().sum()

0

- No duplicates in the data

In [11]:
Q1 = daily_covid_data['New_cases'].quantile(0.25)
Q3 = daily_covid_data['New_cases'].quantile(0.75)
IQR = Q3 - Q1
min = Q1 - 1.5 * IQR
max = Q3 + 1.5 * IQR
daily_covid_data = daily_covid_data[(daily_covid_data['New_cases'] >= min) & (daily_covid_data['New_cases'] <= max)]

- removed outliers from columns 'New_cases' and 'Cumulative_cases' and 'New_deaths' and 'Cumulative_deaths'

In [12]:
Q1 = daily_covid_data['Cumulative_cases'].quantile(0.25)
Q3 = daily_covid_data['Cumulative_cases'].quantile(0.75)
IQR = Q3 - Q1
min = Q1 - 1.5 * IQR
max = Q3 + 1.5 * IQR
daily_covid_data = daily_covid_data[(daily_covid_data['Cumulative_cases'] >= min) & (daily_covid_data['Cumulative_cases'] <= max)]

In [13]:
Q1 = daily_covid_data['New_deaths'].quantile(0.25)
Q3 = daily_covid_data['New_deaths'].quantile(0.75)
IQR = Q3 - Q1
min = Q1 - 1.5 * IQR
max = Q3 + 1.5 * IQR
daily_covid_data = daily_covid_data[(daily_covid_data['New_deaths'] >= min) & (daily_covid_data['New_deaths'] <= max)]

In [14]:
Q1 = daily_covid_data['Cumulative_deaths'].quantile(0.25)
Q3 = daily_covid_data['Cumulative_deaths'].quantile(0.75)
IQR = Q3 - Q1
min = Q1 - 1.5 * IQR
max = Q3 + 1.5 * IQR
daily_covid_data = daily_covid_data[(daily_covid_data['Cumulative_deaths'] >= min) & (daily_covid_data['Cumulative_deaths'] <= max)]

In [15]:
daily_covid_data.describe().T

count          mean            std   min     25%  \
New_cases          11241.0   1253.724402    1830.344519 -88.0   131.0   
Cumulative_cases   11241.0  83638.993239  134724.108525   1.0  6655.0   
New_deaths         11241.0     13.233253      17.527773  -3.0     2.0   
Cumulative_deaths  11241.0   1000.285028    1388.967704   1.0    90.0   

                       50%      75%       max  
New_cases            459.0   1526.0    9961.0  
Cumulative_cases   25383.0  98211.0  982307.0  
New_deaths             5.0     18.0      77.0  
Cumulative_deaths    341.0   1324.0    6185.0

In [16]:
daily_covid_data['New_cases'] = daily_covid_data['New_cases'].clip(lower=0)
daily_covid_data['New_deaths'] = daily_covid_data['New_deaths'].clip(lower=0)

- changing Negative Values in "New_cases" and "New_dearhs" into Zeros cause it doesn't make sense in the context of COVID-19 data.

In [17]:
daily_covid_data.describe().T

count          mean            std  min     25%      50%  \
New_cases          11241.0   1253.732497    1830.338785  0.0   131.0    459.0   
Cumulative_cases   11241.0  83638.993239  134724.108525  1.0  6655.0  25383.0   
New_deaths         11241.0     13.233698      17.527409  0.0     2.0      5.0   
Cumulative_deaths  11241.0   1000.285028    1388.967704  1.0    90.0    341.0   

                       75%       max  
New_cases           1526.0    9961.0  
Cumulative_cases   98211.0  982307.0  
New_deaths            18.0      77.0  
Cumulative_deaths   1324.0    6185.0

In [18]:
daily_covid_data.isna().sum()

Date_reported        0
Country_code         0
Country              0
WHO_region           0
New_cases            0
Cumulative_cases     0
New_deaths           0
Cumulative_deaths    0
dtype: int64

- By removing the Outliers, All the Null values that were located in columns "New_cases" and "New_deaths" were all Outliers so they were removed too

- No Null Values
- No Duplicates
- No Outliers
- No inconsistency
- No Wrong Datatype
- Data is Cleaned

---------------

# Data Cleaning of: Vaccination-data

In [19]:
vaccination_data.head(10)

COUNTRY ISO3 WHO_REGION DATA_SOURCE DATE_UPDATED  \
0                Aruba  ABW       AMRO   REPORTING   29/12/2023   
1          Afghanistan  AFG       EMRO   REPORTING   31/12/2023   
2               Angola  AGO       AFRO   REPORTING   31/12/2023   
3             Anguilla  AIA       AMRO   REPORTING   29/12/2023   
4              Albania  ALB       EURO   REPORTING   23/12/2023   
5              Andorra  AND       EURO   REPORTING   23/12/2023   
6            Argentina  ARG       AMRO   REPORTING   29/12/2023   
7              Armenia  ARM       EURO   REPORTING   23/12/2023   
8  Antigua and Barbuda  ATG       AMRO   REPORTING   29/12/2023   
9            Australia  AUS       WPRO   REPORTING   31/12/2023   

  TOTAL_VACCINATIONS  PERSONS_VACCINATED_1PLUS_DOSE  \
0             217124                        90493.0   
1           2.30E+07                     19200000.0   
2           2.78E+07                     16600000.0   
3              24864                        10858.0   
4            3088966                      1349255.0   
5             157072                        57913.0   
6           1.17E+08                     41200000.0   
7            2256919                      1150915.0   
8             136512                        64290.0   
9           6.97E+07                     22400000.0   

   TOTAL_VACCINATIONS_PER100  PERSONS_VACCINATED_1PLUS_DOSE_PER100  \
0                      203.0                                  85.0   
1                       59.0                                  49.0   
2                       85.0                                  50.0   
3                      166.0                                  72.0   
4                      107.0                                  47.0   
5                      203.0                                  75.0   
6                      258.0                                  91.0   
7                       76.0                                  39.0   
8                      139.0                                  66.0   
9                      273.0                                  88.0   

   PERSONS_LAST_DOSE  PERSONS_LAST_DOSE_PER100  VACCINES_USED  \
0            84363.0                      79.0            NaN   
1         18400000.0                      47.0            NaN   
2          9609080.0                      29.0            NaN   
3            10382.0                      69.0            NaN   
4          1279333.0                      44.0            NaN   
5            53501.0                      69.0            NaN   
6         38100000.0                      84.0            NaN   
7          1030758.0                      35.0            NaN   
8            62384.0                      64.0            NaN   
9         21800000.0                      85.0            NaN   

  FIRST_VACCINE_DATE NUMBER_VACCINES_TYPES_USED  PERSONS_BOOSTER_ADD_DOSE  \
0         17/02/2021                        NaN                   35659.0   
1         22/02/2021                        NaN                 2729940.0   
2         10/03/2021                        NaN                 3067091.0   
3         05/02/2021                        NaN                    3231.0   
4         13/01/2021                        NaN                  402371.0   
5         20/01/2021                        NaN                   43071.0   
6         29/12/2020                        NaN                27700000.0   
7         27/01/2021                        NaN                   81354.0   
8         01/03/2021                        NaN                    9838.0   
9         21/02/2021                        NaN                14400000.0   

   PERSONS_BOOSTER_ADD_DOSE_PER100  Unnamed: 16  
0                             33.0          NaN  
1                              7.0          NaN  
2                              9.0          NaN  
3                             22.0          NaN  
4                             14.0          NaN  
5                             56.0  

In [20]:
vaccination_data.isna().sum()

COUNTRY                                   0
ISO3                                      0
WHO_REGION                                5
DATA_SOURCE                               1
DATE_UPDATED                              4
TOTAL_VACCINATIONS                        6
PERSONS_VACCINATED_1PLUS_DOSE             6
TOTAL_VACCINATIONS_PER100                 8
PERSONS_VACCINATED_1PLUS_DOSE_PER100      8
PERSONS_LAST_DOSE                         6
PERSONS_LAST_DOSE_PER100                  8
VACCINES_USED                           214
FIRST_VACCINE_DATE                       20
NUMBER_VACCINES_TYPES_USED              214
PERSONS_BOOSTER_ADD_DOSE                 21
PERSONS_BOOSTER_ADD_DOSE_PER100          31
Unnamed: 16                             214
dtype: int64

In [21]:
vaccination_data.shape

(215, 17)

In [22]:
vaccination_data['DATA_SOURCE'] = vaccination_data['DATA_SOURCE'].fillna("REPORTING")

- Filling every Null Value with "REPORTING" in column 'DATA_SOURCE' as the rest of the data in the column

In [23]:
vaccination_data['WHO_REGION'] =vaccination_data['WHO_REGION'].fillna('unknown')

- Filling unknown Regions with "unknown"

In [24]:
vaccination_data = vaccination_data.dropna(subset=['DATE_UPDATED'])

- dropping Null values in column "DATE_UPDATED"


In [25]:
vaccination_data = vaccination_data.drop(vaccination_data[vaccination_data['DATE_UPDATED'] == 'REPORTING'].index)

- Droping a specific row cause of inconsistency and not the correct datatype

In [26]:
vaccination_data.describe().T

count          mean           std  \
PERSONS_VACCINATED_1PLUS_DOSE         207.0  2.711717e+07  1.198076e+08   
TOTAL_VACCINATIONS_PER100             206.0  1.543252e+02  8.574355e+01   
PERSONS_VACCINATED_1PLUS_DOSE_PER100  206.0  6.288350e+01  2.406305e+01   
PERSONS_LAST_DOSE                     207.0  2.493210e+07  1.136034e+08   
PERSONS_LAST_DOSE_PER100              206.0  5.786893e+01  2.420537e+01   
VACCINES_USED                           0.0           NaN           NaN   
PERSONS_BOOSTER_ADD_DOSE              193.0  1.298811e+07  6.396833e+07   
PERSONS_BOOSTER_ADD_DOSE_PER100       183.0  3.326776e+01  2.393626e+01   
Unnamed: 16                             0.0           NaN           NaN   

                                         min        25%        50%  \
PERSONS_VACCINATED_1PLUS_DOSE         1638.0  460217.00  3753733.0   
TOTAL_VACCINATIONS_PER100                4.0      82.25      154.5   
PERSONS_VACCINATED_1PLUS_DOSE_PER100     4.0      45.25       67.0   
PERSONS_LAST_DOSE                     1635.0  412207.50  3346176.0   
PERSONS_LAST_DOSE_PER100                 3.0      39.00       62.5   
VACCINES_USED                            NaN        NaN        NaN   
PERSONS_BOOSTER_ADD_DOSE               314.0   76359.00   864723.0   
PERSONS_BOOSTER_ADD_DOSE_PER100          1.0      10.00       31.0   
Unnamed: 16                              NaN        NaN        NaN   

                                              75%           max  
PERSONS_VACCINATED_1PLUS_DOSE         13750000.00  1.320000e+09  
TOTAL_VACCINATIONS_PER100                  222.25  4.700000e+02  
PERSONS_VACCINATED_1PLUS_DOSE_PER100        83.00  1.000000e+02  
PERSONS_LAST_DOSE                     11650000.00  1.280000e+09  
PERSONS_LAST_DOSE_PER100                    77.75  1.000000e+02  
VACCINES_USED                                 NaN           NaN  
PERSONS_BOOSTER_ADD_DOSE               5322496.00  8.340000e+08  
PERSONS_BOOSTER_ADD_DOSE_PER100             56.00  8.300000e+01  
Unnamed: 16                                   NaN           NaN

In [27]:
from scipy.stats.mstats import winsorize

columns_to_check = ['PERSONS_VACCINATED_1PLUS_DOSE', 'TOTAL_VACCINATIONS_PER100','PERSONS_VACCINATED_1PLUS_DOSE_PER100', 'PERSONS_LAST_DOSE', 'PERSONS_LAST_DOSE_PER100', 'PERSONS_BOOSTER_ADD_DOSE', 'PERSONS_BOOSTER_ADD_DOSE_PER100']

if vaccination_data.shape[0] > 0:  # Check if there are any rows
    for col in columns_to_check:
        vaccination_data[col] = winsorize(vaccination_data[col], limits=0.05)

- Checks if the DataFrame has any rows.
- Iterates through each specified column.
- Caps the extreme values in each column using winsorize.
- winsorize: This function is used to cap extreme values in a dataset, effectively handling outliers.
- The limits=0.05 argument specifies that the top and bottom 5% of the data points will be capped. This means that the outliers (the extreme values) will be replaced with the values at the 5th and 95th percentiles.

In [28]:
vaccination_data.describe().T

/opt/conda/lib/python3.10/site-packages/numpy/lib/function_base.py:4824: UserWarning: Warning: 'partition' will ignore the 'mask' of the MaskedArray.
  arr.partition(
/opt/conda/lib/python3.10/site-packages/numpy/lib/function_base.py:4824: UserWarning: Warning: 'partition' will ignore the 'mask' of the MaskedArray.
  arr.partition(
/opt/conda/lib/python3.10/site-packages/numpy/lib/function_base.py:4824: UserWarning: Warning: 'partition' will ignore the 'mask' of the MaskedArray.
  arr.partition(
/opt/conda/lib/python3.10/site-packages/numpy/lib/function_base.py:4824: UserWarning: Warning: 'partition' will ignore the 'mask' of the MaskedArray.
  arr.partition(
/opt/conda/lib/python3.10/site-packages/numpy/lib/function_base.py:4824: UserWarning: Warning: 'partition' will ignore the 'mask' of the MaskedArray.
  arr.partition(
/opt/conda/lib/python3.10/site-packages/pandas/core/construction.py:820: UserWarning: Warning: converting a masked element to nan.
  subarr = np.asarray(arr, dtype=d

count          mean           std  \
PERSONS_VACCINATED_1PLUS_DOSE         207.0  1.398988e+07  2.399692e+07   
TOTAL_VACCINATIONS_PER100             207.0  1.530097e+02  7.915108e+01   
PERSONS_VACCINATED_1PLUS_DOSE_PER100  207.0  6.325121e+01  2.312051e+01   
PERSONS_LAST_DOSE                     207.0  1.261249e+07  2.174719e+07   
PERSONS_LAST_DOSE_PER100              207.0  5.804348e+01  2.325279e+01   
VACCINES_USED                           0.0           NaN           NaN   
PERSONS_BOOSTER_ADD_DOSE              193.0  1.298821e+07  6.396831e+07   
PERSONS_BOOSTER_ADD_DOSE_PER100       183.0  3.332240e+01  2.386477e+01   
Unnamed: 16                             0.0           NaN           NaN   

                                          min       25%        50%  \
PERSONS_VACCINATED_1PLUS_DOSE         25123.0  460217.0  3753733.0   
TOTAL_VACCINATIONS_PER100                29.0      82.5      155.0   
PERSONS_VACCINATED_1PLUS_DOSE_PER100     19.0      45.5       67.0   
PERSONS_LAST_DOSE                     23633.0  412207.5  3346176.0   
PERSONS_LAST_DOSE_PER100                 15.0      39.0       63.0   
VACCINES_USED                             NaN       NaN        NaN   
PERSONS_BOOSTER_ADD_DOSE               3726.0   76359.0   864723.0   
PERSONS_BOOSTER_ADD_DOSE_PER100           3.0      10.0       31.0   
Unnamed: 16                               NaN       NaN        NaN   

                                             75%          max  
PERSONS_VACCINATED_1PLUS_DOSE         13750000.0   90500000.0  
TOTAL_VACCINATIONS_PER100                  223.5        278.0  
PERSONS_VACCINATED_1PLUS_DOSE_PER100        83.0         96.0  
PERSONS_LAST_DOSE                     11650000.0   81300000.0  
PERSONS_LAST_DOSE_PER100                    78.0         92.0  
VACCINES_USED                                NaN          NaN  
PERSONS_BOOSTER_ADD_DOSE               5322496.0  834000000.0  
PERSONS_BOOSTER_ADD_DOSE_PER100             56.0         83.0  
Unnamed: 16                                  NaN          NaN

In [29]:
vaccination_data.duplicated().sum()

0

- No duplicates in the data

In [30]:
vaccination_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 207 entries, 0 to 214
Data columns (total 17 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   COUNTRY                               207 non-null    object 
 1   ISO3                                  207 non-null    object 
 2   WHO_REGION                            207 non-null    object 
 3   DATA_SOURCE                           207 non-null    object 
 4   DATE_UPDATED                          207 non-null    object 
 5   TOTAL_VACCINATIONS                    207 non-null    object 
 6   PERSONS_VACCINATED_1PLUS_DOSE         207 non-null    float64
 7   TOTAL_VACCINATIONS_PER100             207 non-null    float64
 8   PERSONS_VACCINATED_1PLUS_DOSE_PER100  207 non-null    float64
 9   PERSONS_LAST_DOSE                     207 non-null    float64
 10  PERSONS_LAST_DOSE_PER100              207 non-null    float64
 11  VACCINES_USED           

In [31]:
vaccination_data = vaccination_data.drop(vaccination_data[vaccination_data['TOTAL_VACCINATIONS'] == '18/10/2022'].index)

- dropped a specific row in column 'TOTAL_VACCINATIONS' cause of inconsistency and wrong datatype

In [32]:
vaccination_data['TOTAL_VACCINATIONS'] = vaccination_data['TOTAL_VACCINATIONS'].astype(float)

- changed column 'TOTAL_VACCINATIONS' from Object to Float

In [33]:
average_total_vac = vaccination_data['TOTAL_VACCINATIONS'].mean()

vaccination_data['TOTAL_VACCINATIONS'] = vaccination_data['TOTAL_VACCINATIONS'].fillna(average_total_vac)

- filled every Null value in column 'TOTAL_VACCINATIONS' with the Average

In [34]:
vaccination_data = vaccination_data.dropna(subset=['PERSONS_VACCINATED_1PLUS_DOSE_PER100'])

- dropped Null values in column 'PERSONS_VACCINATED_1PLUS_DOSE_PER100'

In [35]:
vaccination_data.isna().sum()

COUNTRY                                   0
ISO3                                      0
WHO_REGION                                0
DATA_SOURCE                               0
DATE_UPDATED                              0
TOTAL_VACCINATIONS                        0
PERSONS_VACCINATED_1PLUS_DOSE             0
TOTAL_VACCINATIONS_PER100                 0
PERSONS_VACCINATED_1PLUS_DOSE_PER100      0
PERSONS_LAST_DOSE                         0
PERSONS_LAST_DOSE_PER100                  0
VACCINES_USED                           207
FIRST_VACCINE_DATE                       12
NUMBER_VACCINES_TYPES_USED              207
PERSONS_BOOSTER_ADD_DOSE                 14
PERSONS_BOOSTER_ADD_DOSE_PER100          24
Unnamed: 16                             207
dtype: int64

In [36]:
vaccination_data = vaccination_data.drop(columns='Unnamed: 16')

- Dropped Column "Unnamed: 16" cause its an empty Column

In [37]:
vaccination_data = vaccination_data.drop(columns='VACCINES_USED')
vaccination_data = vaccination_data.drop(columns='NUMBER_VACCINES_TYPES_USED')

- 'VACCINES_USED' and 'NUMBER_VACCINES_TYPES_USED' are dropped cause no data are in the columns

In [38]:
vaccination_data.isna().sum()

COUNTRY                                  0
ISO3                                     0
WHO_REGION                               0
DATA_SOURCE                              0
DATE_UPDATED                             0
TOTAL_VACCINATIONS                       0
PERSONS_VACCINATED_1PLUS_DOSE            0
TOTAL_VACCINATIONS_PER100                0
PERSONS_VACCINATED_1PLUS_DOSE_PER100     0
PERSONS_LAST_DOSE                        0
PERSONS_LAST_DOSE_PER100                 0
FIRST_VACCINE_DATE                      12
PERSONS_BOOSTER_ADD_DOSE                14
PERSONS_BOOSTER_ADD_DOSE_PER100         24
dtype: int64

In [39]:
vaccination_data.shape

(207, 14)

In [40]:
vaccination_data['FIRST_VACCINE_DATE'] =vaccination_data['FIRST_VACCINE_DATE'].fillna('unknown')

- filled the Null values in column 'FIRST_VACCINE_DATE' with 'unknown'

In [41]:
average_PERSONS_BOOSTER_ADD_DOSE = vaccination_data['PERSONS_BOOSTER_ADD_DOSE'].mean()

vaccination_data['PERSONS_BOOSTER_ADD_DOSE'] = vaccination_data['PERSONS_BOOSTER_ADD_DOSE'].fillna(average_PERSONS_BOOSTER_ADD_DOSE)



average_PERSONS_BOOSTER_ADD_DOSE_PER100 = vaccination_data['PERSONS_BOOSTER_ADD_DOSE_PER100'].mean()

vaccination_data['PERSONS_BOOSTER_ADD_DOSE_PER100'] = vaccination_data['PERSONS_BOOSTER_ADD_DOSE_PER100'].fillna(average_PERSONS_BOOSTER_ADD_DOSE_PER100)

- filled the null values in columns 'PERSONS_BOOSTER_ADD_DOSE' and 'PERSONS_BOOSTER_ADD_DOSE_PER100' with the average for each column

In [42]:
vaccination_data.isna().sum()

COUNTRY                                 0
ISO3                                    0
WHO_REGION                              0
DATA_SOURCE                             0
DATE_UPDATED                            0
TOTAL_VACCINATIONS                      0
PERSONS_VACCINATED_1PLUS_DOSE           0
TOTAL_VACCINATIONS_PER100               0
PERSONS_VACCINATED_1PLUS_DOSE_PER100    0
PERSONS_LAST_DOSE                       0
PERSONS_LAST_DOSE_PER100                0
FIRST_VACCINE_DATE                      0
PERSONS_BOOSTER_ADD_DOSE                0
PERSONS_BOOSTER_ADD_DOSE_PER100         0
dtype: int64

- No Null values
- No duplicates
- No inconsistency
- No wrong datatypes
- Outliers are removed
- All columns are cleaned

# Data Analysis Part:-